src/main/java/com/puppycrawl/tools/checkstyle/utils/AnnotationUtil.java

```diff
@@ -167,4 +169,31 @@ public static DetailAST getAnnotation(final DetailAST ast,
         return result;
     }
 
+    /**
+     * Checks if the given AST element is marked with an annotation that allows
+     * it to omit javadoc documentation.
+     * @param ast The type or method definition.
+     * @param allowedAnnotations A collection of annotations that allow omitting documentation.
+     * @return true if the type or method is exempt from comments, false otherwise.
+     */
+    public static boolean hasAllowedAnnotations(DetailAST ast, List<String> allowedAnnotations) {
+        boolean result = false;
+        final DetailAST annotationHolderNode = getAnnotationHolder(ast);
+        DetailAST annotationNode = annotationHolderNode.findFirstToken(TokenTypes.ANNOTATION);
+        while (annotationNode != null && annotationNode.getType() == TokenTypes.ANNOTATION) {
```

Comment:
> annotationNode.getType() == TokenTypes.ANNOTATION

I don't think you can have this here, since Javadocs/comments sometimes get placed inside the `modifiers` section next to annotations.
Add some javadocs to your tests to confirm.

Reply:
I moved this code from its original location, so I have not changed this at all.
Could you please give me an example of what you mean so I can add the test?

Reply:
@rnveach , please reply.

Reply:
pending reply from @rnveach